src/core/openjpeg_decoder.js

```diff
@@ -0,0 +1,102 @@
+/* Copyright 2024 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { BaseException } from "../shared/util.js";
+import OpenJPEG from "../../external/openjpeg/openjpeg.js";
+
+class JpxError extends BaseException {
+  constructor(msg) {
+    super(`JPX error: ${msg}`, "JpxError");
+  }
+}
+
+class OpenJPEGDecoder {
```

Comment:
Now that we rely on the upstream OpenJPEG implementation, we might be able to remove the JPX fuzzer (see https://github.com/mozilla/pdf.js/blob/master/test/fuzz/jpx_image.fuzz.js and https://github.com/google/oss-fuzz/blob/master/projects/pdf-js/build.sh#L25 because we no longer use our own decoder. It looks like OpenJPEG is already fuzzed (see https://github.com/uclouvain/openjpeg/tree/master/tests/fuzzers and https://github.com/google/oss-fuzz/tree/master/projects/openjpeg) but it's not entirely clear to me if JPX is part of that (maybe @manunio can chime in here given that they introduced our fuzzers in #17337 and might know more about this).

Reply:
Coming to think of this again, the https://github.com/mozilla/pdf.js.openjpeg repository does contain some wrapping code for OpenJPEG which I guess could be useful to fuzz. I'm not sure how useful that is given that OpenJPEG itself is already fuzzed upstream and we're only wrapping it thinly, but if it is we might not actually have to remove the fuzzer because it's then still fuzzing some of our code (perhaps we only need to update the exception message from the other comment).